### PR TITLE
Add MIO keyboard

### PR DESCRIPTION
**Pull request by takai**
## Description

Add MIO keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mio/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make mio:default
+
+Flashing example for this keyboard:
+
+    make mio:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (drashna) [inline keyboards/mio/readme.md:19]: Could you include instructions on how to enter the bootloader?

Comment (takai) [reply keyboards/mio/readme.md:19]: Thank you! Updated 9af05b1de9627ac3198bcd23ae4fa8d53c6665a3

**Review by drashna — APPROVED** [review]

File: keyboards/mio/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        _______, _______, _______, _______, _______, _______, _______,          \
+        _______, _______, _______,                   _______, _______           \
+    ),
+    [3] = LAYOUT(
+        _______,          _______, _______, _______, _______, _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______,          \
+        _______, _______, _______, _______, _______, _______, _______,          \
+        _______, _______, _______, _______, _______, _______, _______,          \
+        _______, _______, _______,                   _______, _______           \
```
Comment (fauxpark) [inline keyboards/mio/keymaps/via/keymap.c:49]:
```suggestion
    [0] = LAYOUT(
        KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,
        MO(1),   KC_LCTL, KC_LALT,                   KC_SPC,  KC_ENT
    ),
    [1] = LAYOUT(
        RESET,             _______, _______, _______, _______, _______, _______,
        RGB_TOG, RGB_MOD,  RGB_HUI, RGB_SAI, RGB_VAI, _______, _______, _______,
        _______, RGB_RMOD, RGB_HUD, RGB_SAD, RGB_VAD, _______, _______,
        _______, _______,  _______, _______, _______, _______, _______,
        _______, _______,  _______, _______, _______, _______, _______,
        _______, _______,  _______,                   _______, _______
    ),
    [2] = LAYOUT(
        _______,          _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______,                   _______, _______
    ),
    [3] = LAYOUT(
        _______,          _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______,                   _______, _______
```

File: keyboards/mio/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,           \
+        MO(_FN), KC_LCTL, KC_LALT,                   KC_SPC,  KC_ENT          \
+    ),
+    [_FN] = LAYOUT(
+        RESET,             _______, _______, _______, _______, _______, _______, \
+        RGB_TOG, RGB_MOD,  RGB_HUI, RGB_SAI, RGB_VAI, _______, _______, _______, \
+        _______, RGB_RMOD, RGB_HUD, RGB_SAD, RGB_VAD, _______, _______,          \
+        _______, _______,  _______, _______, _______, _______, _______,          \
+        _______, _______,  _______, _______, _______, _______, _______,          \
+        _______, _______, _______,                    _______, _______           \
```
Comment (fauxpark) [inline keyboards/mio/keymaps/default/keymap.c:40]:
```suggestion
        KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,
        MO(_FN), KC_LCTL, KC_LALT,                   KC_SPC,  KC_ENT
    ),
    [_FN] = LAYOUT(
        RESET,             _______, _______, _______, _______, _______, _______,
        RGB_TOG, RGB_MOD,  RGB_HUI, RGB_SAI, RGB_VAI, _______, _______, _______,
        _______, RGB_RMOD, RGB_HUD, RGB_SAD, RGB_VAD, _______, _______,
        _______, _______,  _______, _______, _______, _______, _______,
        _______, _______,  _______, _______, _______, _______, _______,
        _______, _______, _______,                    _______, _______
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**